init/include/hpx/hpx_main.hpp

```diff
@@ -18,13 +18,17 @@
     (defined(__linux) || defined(__linux__) || defined(linux) || \
     defined(__APPLE__))
 
+#include <string>
+
 namespace hpx_start
 {
     // include_libhpx_wrap here is an override for the one present in
     // src/hpx_wrap.cpp. The value of this variable defines if we need
     // to change the program's entry point or not.
     HPX_SYMBOL_EXPORT extern bool include_libhpx_wrap;
     HPX_SYMBOL_EXPORT bool include_libhpx_wrap = true;
+    HPX_SYMBOL_EXPORT extern std::string app_name_libhpx_wrap;
+    HPX_SYMBOL_EXPORT std::string app_name_libhpx_wrap = HPX_APPLICATION_STRING;
```

Comment:
Let's make this `constexpr` instead:
```
constexpr char const app_name_libhpx_wrap[] = HPX_APPLICATION_STRING;
```

Reply:
I'm not fully understanding what's going on with the hpx_wrap thing but when I just put the line you suggested, `app_name_libhpx_wrap` is then undefined in `src/hpx_wrap.cpp`

The problem is that the for certain tests which use hpx_wrap, the `HPX_APPLICATION_STRING` is defined in `hpx/hpx_main.hpp` and set to "Unknown HPX application", my goal was to use the same method with `__attribute__((weak))` symbols to make it overwrite with the right application name but the `__attribute__((weak))` trick seems to work only with gcc, I really need help on that one..